src/poetry/utils/authenticator.py

```diff
@@ -267,6 +269,19 @@ def _get_credentials_for_url(self, url: str) -> HTTPAuthCredential:
 
         return credential
 
+    def get_credentials_for_git_url(self, url: str) -> HTTPAuthCredential:
+        parsed_url = urllib.parse.urlsplit(url)
+
+        if parsed_url.scheme not in {"http", "https"}:
+            return HTTPAuthCredential()
```

Comment:
Is there a good reason to return an empty credential over None?

Reply:
Additional guards on consumers is one reason. And this is a bit more cleaner than returning none here imho. It also keeps interfaces uniform when interacting with the authenticator.